client/src/components/FilterForm.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import { Search as SearchIcon } from 'lucide-react';
+
+type FilterProps = {
+  onSearchInput: (value: string) => void;
+};
+
+export default function FilterForm({ onSearchInput }: FilterProps) {
+  const [selectOption, setSelectOption] = useState({
+    location: '',
+    level: '',
+    salary: '',
```

Comment:
This is OK, but as you can see it can be a bit of a nuisance to set the state when the selection changes. An alternative would be to have 3 separate `useState` variables.

It doesn't matter which one you use. Just giving you an option to consider.